src/main/java/nextstep/subway/application/strategy/addition/AddSectionAfterUpStationStrategy.java

```diff
@@ -24,10 +22,10 @@ private boolean areOnlyUpStationsSame(Section existingSection, Section newSectio
     }
 
     @Override
-    public void addSection(Line line, List<Section> sections, Section newSection) {
-        Section existingSection = findExistingSectionForNewAddition(sections, newSection);
+    public void addSection(Line line, Section newSection) {
+        Section existingSection = findExistingSectionForNewAddition(line.getSections(), newSection).get();
```

Comment:
```suggestion
        Section existingSection = findExistingSectionForNewAddition(line.getSections(), newSection).orElseThrow(~~~);
```
`canApply` 를 통과한 상태라 `Optional.empty()` 라면 예외가 발생하도록 하는 게 맞을 것 같아요 🤔 
실제로도 `Optional.empty()` 상태라면 `NoSuchElementException` 가 발생하기 때문에, 사용자 정의 예외를 넣어 주는 선택이 더 좋다고 생각합니다